src/main/java/com/aws/greengrass/deployment/DeploymentService.java

```diff
@@ -504,10 +505,12 @@ private void cancelCurrentDeployment() {
                     .isCancellable()) {
                 logger.atInfo().log("Deployment already finished processing or cannot be cancelled");
             } else {
-                boolean canCancelDeployment = context.get(UpdateSystemPolicyService.class).discardPendingUpdateAction(
```

Comment:
Currently, this returns `true` for the deployments that do not have the update policy set to  `DeploymentComponentUpdatePolicyAction.NOTIFY_COMPONENTS` which is wrong. We should discarding an update only when the policy is set to notify.